### PR TITLE
Propose a backend working group

### PR DESCRIPTION
**Pull request by russellb**
There is a broad group of people interested in collaborating on the
software needed to scale out the workflow in a clustered environment.
As a first step toward formalizing this work, this document describes
a working group that will collaborate on documents in this repository
to defined high level architecture and design for the related
components to build out this updated backend infrastructure.

Signed-off-by: Russell Bryant <rbryant@redhat.com>

### CODE REVIEW

**Comment by russellb** [comment]
Rendered version of the document: https://github.com/russellb/instructlab-dev-docs/tree/backend/docs/backend

File: docs/backend/README.md
Comment (bjhargrave) [inline docs/backend/README.md:45]:
```suggestion
- **Slack**: The group uses the `#backend` channel in the InstructLab Slack.
```

Shouldn't this just be plain "backend" in alignment with other channel names?

**Review by bjhargrave — COMMENTED** [review]
Maybe add a label to this repo for "backend" things?

File: docs/backend/README.md
Comment (russellb) [reply docs/backend/README.md:45]: Yes!  That's a typo. The `#backend` channel already exists, even.

**Comment by russellb** [comment]
> Maybe add a label to this repo for "backend" things?

Great suggestion! I'll go ahead and create the label. Later we could configure Mergify to auto label any PRs that touch this directory.

File: docs/backend/README.md
Comment (russellb) [reply docs/backend/README.md]: fixed, thanks!

**Review by gabe-l-hart — COMMENTED** [review]
Thanks for putting this together @russellb! Is it worth defining the initial members of the working group so that viewers can use it as a lookup?

**Comment by russellb** [comment]
> Thanks for putting this together @russellb! Is it worth defining the initial members of the working group so that viewers can use it as a lookup?

My proposal for group membership to start was "just show up and participate" :) .  The members of the `#backend` slack channel would somewhat serve as a membership list.

There's a related "who approves the group's artifacts" ? That is covered under the policy for this repo in general, which is discussed in the top-level README here: https://github.com/instructlab/dev-docs

It comes down to an Oversight Committee member doing the final approval and merge of docs. The O.C. member would be responsible for reviewing the PR to ensure adequate consensus has been reached and that adequate opportunity for input has been provided.

The O.C. membership is listed in this doc: https://github.com/instructlab/community/blob/main/MAINTAINERS.md

Once this work gets far enough to create some code repositories, we would create one or more new Maintainer teams responsible for review and approval of changes to those code repos.

Does that all sound reasonable?

**Review by mairin — APPROVED** [review]
LGTM

File: docs/backend/README.md
Comment (leseb) [inline docs/backend/README.md:32]: Where should the implementation (actual code) live? Is it TBD?

**Comment by lhawthorn** [comment]
LGTM, modulo the clarification requested by @leseb

**Review by nathan-weinberg — APPROVED** [review]

**Review by danmcp — APPROVED** [review]
LGTM

File: docs/backend/README.md
Comment (russellb) [reply docs/backend/README.md:32]: TBD -- I would expect that to be laid out in some of the design docs created by the group and placed in this new directory.

**Comment by russellb** [comment]
This appears to have broad support with several approvals, so I'm going to merge now. Thanks!

**Pull request merged**
